lib/datasource/remote/model/token_model.dart

```diff
@@ -113,8 +117,37 @@ class TokenModel extends Equatable {
   @override
   List<Object?> get props => [chainName, contract, symbol];
 
-  String getAssetString(double quantity) {
-    return "${quantity.toStringAsFixed(precision)} $symbol";
+  static String getAssetString(String? id, double quantity) {
+    if (id!=null && TokenModel.fromId(id)!=null && contractPrecisions.containsKey(id)) {
+      final symbol = TokenModel.fromId(id)!.symbol;
+      return symbol==null ? "" : "${quantity.toStringAsFixed(contractPrecisions[id]!)} $symbol";
+    } else {
+      return "";
+    }
+  }
+
+  void setPrecisionFromString(String s) {
+    final amount = s.split(' ')[0];
+    final ss = amount.split('.');
+    if (ss.isEmpty) {
+      return;
+    }
+    contractPrecisions[this.id] = ss.length==1 ? 0 : ss[1].length;
+  }
+
+  // enabling 'send' transfer validity checks, e.g. Mutual Credit,
+  //  membership limitations
```

Comment:
What does this method do and why is it called what it is called??

Needs better name?

Reply:
This is about warning the user in advance that their transaction won't go through, rather than waiting for the contract to reject it. It hasn't been exercised because so far no token has provided the matching token metadata (overdraw flag) in the tmastr.seeds database. So it's not doing anything at the moment.